pkg/porter/publish.go

```diff
@@ -1,55 +1,53 @@
 package porter
 
 import (
+	"bytes"
 	"context"
 	"fmt"
 	"os"
 	"path/filepath"
 	"strings"
 
+	"github.com/deislabs/cnab-go/bundle"
 	"github.com/deislabs/porter/pkg/config"
 	"github.com/docker/cli/cli/command"
+	dockerconfig "github.com/docker/cli/cli/config"
 	cliflags "github.com/docker/cli/cli/flags"
 	"github.com/docker/distribution/reference"
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/pkg/jsonmessage"
 	"github.com/docker/docker/pkg/term"
 	"github.com/docker/docker/registry"
+
+	"github.com/docker/cnab-to-oci/remotes"
 	"github.com/pkg/errors"
 )
 
 // PublishOptions are options that may be specified when publishing a bundle.
 // Porter handles defaulting any missing values.
 type PublishOptions struct {
-	File     string
-	Tag      string
-	CNABFile string
+	File             string
+	InsecureRegistry bool
 }
 
+// Validate performs validation on the publish options
 func (p PublishOptions) Validate(porter *Porter) error {
-	if p.File == "" {
-		fs := porter.Context.FileSystem
+	fs := porter.Context.FileSystem
+	f := p.File
+	if f == "" {
 		pwd, err := os.Getwd()
 		if err != nil {
 			return errors.Wrap(err, "could not get current working directory")
 		}
+		f = filepath.Join(pwd, config.Name)
```

Comment:
I notice that we are duplicating validation between sharedOptions and here. As part of my --cnab-file work, I'm making another struct that is going to extract out logic for --file and --cnab-file that will do the common validation for these flags across all the commands. Just a heads up.